src/test/lib/optimizer/strategy/predicate_reordering_rule_test.cpp

```diff
@@ -269,4 +262,59 @@ TEST_F(PredicateReorderingTest, SimpleValidateReorderingTest) {
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
 }
 
+TEST_F(PredicateReorderingTest, DoNotReorderMultiPredicateSemiAndAntiJoins) {
+  // The semi-/anti-joins filter the left input relation, the PredicateNode does not. However, do not reorder the nodes
+  // as we cannot execute multi-predicate joins efficiently.
+  const auto node_b = static_pointer_cast<MockNode>(node->deep_copy());
+  const auto b_a = node_b->get_column("a");
+  const auto b_b = node_b->get_column("b");
+  const auto b_c = node_b->get_column("c");
+
+  for (const auto join_mode : {JoinMode::Semi, JoinMode::AntiNullAsTrue, JoinMode::AntiNullAsFalse}) {
+    // clang-format off
+    const auto lqp =
+    JoinNode::make(join_mode, expression_vector(equals_(a, b_a), not_equals_(c, b_c)),
+      PredicateNode::make(greater_than_(b, 0),
+        node),
+      PredicateNode::make(greater_than_(b_a, 50),
+        node_b));
+    // clang-format on
+
+    const auto expected_lqp = lqp->deep_copy();
+
+    apply_rule(_rule, lqp);
+    EXPECT_LQP_EQ(lqp, expected_lqp);
+  }
+}
+
+TEST_F(PredicateReorderingTest, PreferPredicatesOverJoins) {
+  // The PredicateNode has a worse selectivity than the JoinNode. However, we place it earlier in the plan since a scan
+  // is cheaper than a semi-join.
```

Comment:
That's a cost-based decision that might change as soon as we change some magic numbers, right?

Reply:
Yes, absolutely. I assume if someone touches the cost/cardinality estimation, they know what they do and might as well adapt this test case if it does not pass anymore